File: native/libdbusmenu/tests/test-glib-simple-items.c
#include <libdbusmenu-glib/server.h>
#include <libdbusmenu-glib/menuitem.h>

static DbusmenuMenuitem * root_menuitem = NULL;
static GMainLoop * mainloop = NULL;

gchar * dummies[] = {
	"Bob", "Jim", "Alvin", "Mary", NULL
};

static void
dummy_users (DbusmenuMenuitem * root) {
	int count;
	for (count = 0; dummies[count] != NULL; count++) {
		DbusmenuMenuitem * mi = dbusmenu_menuitem_new();
		g_debug("Creating item: %d %s", dbusmenu_menuitem_get_id(mi), dummies[count]);
		g_debug("\tRoot ID: %d", dbusmenu_menuitem_get_id(root));
		dbusmenu_menuitem_property_set(mi, "label", dummies[count]);
		dbusmenu_menuitem_child_add_position(root, mi, count);
	}

	return;
}

static gboolean
quititall (gpointer data)
{
	g_main_loop_quit(mainloop);
	return FALSE;
}

int
main (int argc, char ** argv)
{
    DbusmenuServer * server = dbusmenu_server_new("/test/object");
    root_menuitem = dbusmenu_menuitem_new();
    dbusmenu_server_set_root(server, root_menuitem);
	g_debug("Root ID: %d", dbusmenu_menuitem_get_id(root_menuitem));

	dummy_users(root_menuitem);

	g_timeout_add_seconds(1, quititall, NULL);

    mainloop = g_main_loop_new(NULL, FALSE);
    g_main_loop_run(mainloop);

    return 0;
}

File: native/libdbusmenu/tests/test-glib-simple-items.py
#!/usr/bin/python
# This is the Python GI version of test-glib-simple-items.c

import gobject
from gi.repository import Dbusmenu

dummies = ['Bob', 'Jim', 'Alvin', 'Mary']

def dummy_users(root):
    count = 0
    for user in dummies:
        mi = Dbusmenu.Menuitem()
        print 'Creating item: %d %s' % (mi.get_id(), user)
        print '\tRoot ID:', root.get_id()
        mi.property_set('label', user)
        root.child_add_position(mi, count)
        assert mi.property_get('label') == user
        count += 1

def quititall(mainloop):
    mainloop.quit()
    return False

# main

server = Dbusmenu.Server.new('/test/object')
root_menuitem = Dbusmenu.Menuitem()
server.set_root(root_menuitem)
print 'Root ID:', root_menuitem.get_id()

dummy_users(root_menuitem)

mainloop = gobject.MainLoop()
gobject.timeout_add_seconds(1, quititall, mainloop)
mainloop.run()
